mindsdb/mindsdb
# adding post vector search ranking

## Description

adding post vector search reranking and filtering in order to increase accuracy of the vector search


## Type of change

(Please delete options that are not relevant)


## Verification Process

To ensure the changes are working as expected:


## Additional Media:


## Checklist:


path to the additional unit test:
tests/integrations/utilities/rag/rerankers/test_openai_reranker.py

steps to test end-to-end:
1. start mindsdb locally
2. run "using agents with retrieval" from  mindsdb_python_sdk
3. check that output of the script is correct
